src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,49 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int AMOUNT_OF_FIGURES = 5;
+    private static final int MAX_RANDOM_VALUE = 100;
+    private Random random = new Random();
+    private int numOfFigure;
+    private String color;
+    private ColorSupplier colorSupplier;
+
+    public Figure getRandomFigure() {
+        Figure figure;
+        numOfFigure = random.nextInt(AMOUNT_OF_FIGURES);
+        colorSupplier = new ColorSupplier();
+        color = colorSupplier.getRandomColor();
+        switch (numOfFigure) {
+            case 0:
+                figure = new Rectangle("Rectangle", color, random.nextInt(MAX_RANDOM_VALUE) + 1,
+                        random.nextInt(MAX_RANDOM_VALUE) + 1);
+                break;

```

Comment:
If we make instant return of required figure we will not need figure variable and will not require breaks in our switch (same for all cases)
```suggestion
                return new Rectangle("Rectangle", color, random.nextInt(MAX_RANDOM_VALUE) + 1,
                        random.nextInt(MAX_RANDOM_VALUE) + 1);
```